datastore/importers/nmdb-import-cisco/Parser.test.cpp

```diff
@@ -0,0 +1,829 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// =============================================================================
+// Maintained by Sandia National Laboratories <Netmeld@sandia.gov>
+// =============================================================================
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MAIN
+#include <boost/test/unit_test.hpp>
+
+#include <netmeld/datastore/parsers/ParserTestHelper.hpp>
+
+#include "Parser.hpp"
+
+namespace nmdp = netmeld::datastore::parsers;
+
+using qi::ascii::blank;
+
+class TestParser : public Parser {
+
+  public:
+    using Parser::nocdp;
+    using Parser::versionPIXASA;
+    using Parser::spanningTreeInitial;
+    using Parser::deviceAAA;
+    using Parser::globalServices;
+    using Parser::domainData;
+    using Parser::vrfInstance;
+    using Parser::channelGroup;
+    using Parser::encapsulation;
+    using Parser::switchport;
+    using Parser::switchportPortSecurity;
+    using Parser::switchportVlan;
+    using Parser::spanningTree;
+    using Parser::interface;
+    using Parser::routerId;
+    using Parser::route;
+    using Parser::vlan;
+    using Parser::accessPolicyRelated;
+    using Parser::policyMap;
+    using Parser::classMap;
+
+};
+
+BOOST_AUTO_TEST_CASE(testParts)
+{
+  TestParser tp;
+
+  {
+    const auto& parserRule {tp.nocdp};
+
+    const std::vector<std::string> nocdpTestDataOk
+      {
+        "no cdp run\n",
+        "no cdp enable\n"
+      }
+      ;
+
+    const std::vector<std::string> nocdpTestDataBad
+      {
+        "no cdp\n",
+        "no cdp invalid\n"
+      }
+      ;
+
+    for (const auto& test : nocdpTestDataOk) {
+      BOOST_TEST(nmdp::test(test, parserRule, blank),
+                 "Parse rule 'nocdp':" << test);
+    }
+
+    for (const auto& test : nocdpTestDataBad) {
+      BOOST_TEST(!nmdp::test(test, parserRule, blank),
+                 "Parse rule '!nocdp':" << test);
+    }
+  }
```

Comment:
Primarily, this should ensure `globalCdpEndabled` is not accidentally flipped to `false`.  Again, this comment generally applies throughout this file.